src/main/java/com/mongodb/kafka/connect/util/FlexibleDateTimeParser.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kafka.connect.util;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+import java.time.temporal.TemporalAccessor;
+
+public class FlexibleDateTimeParser {
+
+  public static final String DEFAULT_DATE_TIME_FORMATTER_PATTERN =
+      "yyyy-MM-dd['T'][ ][HH:mm:ss[[.][SSSSSS][SSS]][ ]VV[ ]'['VV']'][HH:mm:ss[[.][SSSSSS][SSS][ ]]X][HH:mm:ss[[.][SSSSSS][SSS]]";
```

Comment:
This pattern evolved to look similar to a regular expression in its readability :) I noticed that it contains 21 `[` and 20 `]`, which means one `]` is missing (it is unfortunate that `DateTimeFormatter` does not thoroughly check that the patten is well-formed). If we format the pattern for readability, it is easy to spot the missing bracket:

```java
"yyyy-MM-dd['T'][ ]"
+ "["
  + "HH:mm:ss" + "[" + "[.][SSSSSS][SSS]" + "]" + "[ ]VV[ ]'['VV']'"
+ "]"
+ "["
  + "HH:mm:ss" + "[" + "[.][SSSSSS][SSS][ ]" + "]" + "X"
+ "]"
+ "["
  + "HH:mm:ss" + "[" + "[.][SSSSSS][SSS]" + "]"
//']' is missing
;
```

Btw, what do you think about formatting the pattern in the code in a similar (not necessary the same way) to improve readability?

Reply:
Great catch and idea, it is a funky pattern :) 

There is auto formatting so I've done the best I can and hope its still readable.